src/cascadia/WindowsTerminal/icon.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "resource.h"
+
+#include "winrt/Windows.ApplicationModel.h"
+
+// clang-format off
+enum class IconClass : uint8_t {
+    None             = 0,
+    Version_Pre      = 0b00000001,
+    Version_Dev      = 0b00000010,
+    Variant_HC       = 0b00000100,
+    Variant_HC_White = 0b00001000,
+};
+
+DEFINE_ENUM_FLAG_OPERATORS(IconClass);
+
+static constexpr std::array<std::pair<IconClass, int>, 9> s_iconClassMapping{
+    std::pair{ IconClass::None,                                                              IDI_APPICON },
+    std::pair{ IconClass::Version_Pre,                                                       IDI_APPICON_PRE },
+    std::pair{ IconClass::Version_Dev,                                                       IDI_APPICON_DEV },
+
+    // High Contrast
+    std::pair{ IconClass::Variant_HC,                                                        IDI_APPICON_HC_B },
+    std::pair{ IconClass::Variant_HC | IconClass::Variant_HC_White,                          IDI_APPICON_HC_W },
+    std::pair{ IconClass::Version_Pre | IconClass::Variant_HC,                               IDI_APPICON_PRE_HC_B },
+    std::pair{ IconClass::Version_Pre | IconClass::Variant_HC | IconClass::Variant_HC_White, IDI_APPICON_PRE_HC_W },
+    std::pair{ IconClass::Version_Dev | IconClass::Variant_HC,                               IDI_APPICON_DEV_HC_B },
+    std::pair{ IconClass::Version_Dev | IconClass::Variant_HC | IconClass::Variant_HC_White, IDI_APPICON_DEV_HC_W },
+};
+// clang-format on
+
+static wchar_t* _GetActiveAppIcon()
+{
+    auto iconClass{ IconClass::None };
+    try
+    {
+        const auto package{ winrt::Windows::ApplicationModel::Package::Current() };
+        const auto id{ package.Id() };
+        const auto name{ id.FullName() };
+        const std::wstring_view nameView{ name };
+        WI_UpdateFlag(iconClass, IconClass::Version_Pre, nameView.find(L"Preview") != std::wstring_view::npos);
+        WI_UpdateFlag(iconClass, IconClass::Version_Dev, nameView.find(L"Dev") != std::wstring_view::npos);
+    }
+    catch (...)
+    {
+        // Just fall through and assume that we're un-badged.
+    }
+
+    HIGHCONTRASTW hcInfo{};
+    hcInfo.cbSize = sizeof(hcInfo);
+
+    if (SystemParametersInfoW(SPI_GETHIGHCONTRAST, sizeof(hcInfo), &hcInfo, 0))
+    {
+        if (WI_IsFlagSet(hcInfo.dwFlags, HCF_HIGHCONTRASTON))
+        {
+            WI_SetFlag(iconClass, IconClass::Variant_HC);
+
+            if (hcInfo.lpszDefaultScheme)

```

Comment:
i'm sure you know this doesn't localize well.  instead, you can use this approach:

```
if (0x00FFFFFF == GetSysColor(COLOR_WINDOW)) {
    WI_SetFlag(HC_White);
}
```
(internally, grep for SystemThemingInterop::GetSystemHighContrastTheme)

I'm not sure if XAML theme resources are available in this part of the application, but if they are, you could alternatively let the XAML theming decide the right theme for you:
```
            <ResourceDictionary.ThemeDictionaries>
                <ResourceDictionary x:Key="Default">
                    <x:Int32 x:Key="Win32IconResourceId">101</x:Int32>
                </ResourceDictionary>
                <ResourceDictionary x:Key="HighContrastBlack">
                    <x:Int32 x:Key="Win32IconResourceId">102</x:Int32>
                </ResourceDictionary>
                <ResourceDictionary x:Key="HighContrastWhite">
                    <x:Int32 x:Key="Win32IconResourceId">103</x:Int32>
                </ResourceDictionary>
            </ResourceDictionary.ThemeDictionaries>
        </ResourceDictionary>
```
😹

Reply:
I wish we had access to theme resources here, but alas.. we don't. ☹️

Good call with the window color. I prefer that over the theme name hack.